packages/async-rewriter/src/symbol-table.ts

```diff
@@ -84,13 +85,24 @@ export default class SymbolTable {
     return this.signatures.unknown;
   }
 
+  private checkIfApi(key, item?): void {
+    if (item === undefined) {
+      item = this.lookup(key);
+    }
+    if (item.api) {
+      // waiting on new error types
+      throw new Error(`InvalidInput - cannot reassign API type ${key}`);
```

Comment:
FYI: this job can be done without the rewriting when we assign the global context in the shell-evaluator:

```
Object.defineProperty(contextObject, 'db', {
  get: () => mapper.currentDatabase,
  writable: false
});
```

Reply:
lol! You're so right, since we save the state before/after js errors. Thank you!

Reply:
Hmm, actually, that won't error (it will just prevent it from being changed). I will look into how to make it error

Reply:
For the moment, until we can figure out the segfault, going to keep the readonly check here